From 2825824b2d9ce17249a7a2bd1a30aa502261ff08 Mon Sep 17 00:00:00 2001
From: Pandhariix <mbusy.pro@gmail.com>
Date: Mon, 16 Dec 2019 16:58:30 +0100
Subject: [PATCH 3/3] Add timestamp to the LogMessage structure for backward
 compatibility

---
 libqicore/qicore/logmessage.hpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libqicore/qicore/logmessage.hpp b/libqicore/qicore/logmessage.hpp
index a36ff2e..8ab36a1 100644
--- a/libqicore/qicore/logmessage.hpp
+++ b/libqicore/qicore/logmessage.hpp
@@ -43,6 +43,9 @@ struct LogMessage
   unsigned int id = 0;                    // Unique message ID
   qi::Clock::time_point date;             // Steady clock timestamp
   qi::SystemClock::time_point systemDate; // Wall clock timestamp
+
+  // timestamp when the message has been posted
+  qi::os::timeval timestamp = qi::os::timeval(systemDate.time_since_epoch());
 };
 }
 
-- 
2.7.4

